### PR TITLE
speedup TriMesh.edgepaths by a factor 20

### PR DESCRIPTION
**Pull request by basnijholt**
Based on what I did for https://github.com/python-adaptive/adaptive/commit/9f481562b0118199d4011667e857ed8eb764a2a5 I made these changes here too.

The numbers say everything :)
```python
# with this PR
%timeit trimesh.edgepaths 
4.7 ms ± 271 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

# @ master
%timeit trimesh.edgepaths
86.6 ms ± 5.65 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

### CODE REVIEW

File: holoviews/element/graphs.py [hunk 1]
```diff
@@ -493,12 +493,15 @@ def edgepaths(self):
...
 
+
         simplices = self.array([0, 1, 2]).astype(np.int32)
         pts = self.nodes.array([0, 1]).astype(float)
-        empty = np.array([[np.NaN, np.NaN]])
-        paths = [arr for tri in pts[simplices] for arr in
-                 (tri[[0, 1, 2, 0], :], empty)][:-1]
-        edgepaths = self.edge_type([np.concatenate(paths)],
+        pts = pts[simplices]
+        paths = np.pad(pts[:, [0, 1, 2, 0], :],
```
Comment (philippjfr) [inline holoviews/element/graphs.py]: I didn't know about np.pad, this is amazing!

**Comment by jlstevens** [comment]
Nice!!

**Comment by philippjfr** [comment]
There's a few other places we can use this trick. I'd be happy to work on those but I'll list the ones I can think of immediately:

[Datashader path concatenation](https://github.com/ioam/holoviews/blob/master/holoviews/operation/datashader.py#L250)

[Chord path concatenation](https://github.com/ioam/holoviews/blob/master/holoviews/element/graphs.py#L621)

**Comment by philippjfr** [comment]
Looks good and tests are passing, merging.

**Pull request merged**